### PR TITLE
feat(dot/network/messages): define `StateRequest` and `StateResponse` proto messages

### LINKED ISSUE #4116: Implement state request protobuf messages
### Description

- Gossamer needs to have the `State` request and response protobuf messages.
- The polkadot spec does not the request body defined so, the definition can be found at: https://github.com/paritytech/polkadot-sdk/blob/eb0a9e593fb6a0f2bbfdb75602a51f4923995529/substrate/client/network/sync/src/schema/api.v1.proto#L73
- The need for this messages is to enable Gossamer sync up to the most up to date state once warp sync process finishes.

### PR DESCRIPTION
**Pull request by EclesioMeloJunior**
## Changes

- In order to execute state requests Gossamer should define the proto definitions for the `StateRequest` and to receive the response to have the proto definition for `StateResponse`. This p2p message is also needed for WARP sync work

- This PR also creates a new package called `messages` under `network` so we can split different messages over there and not in a single `messages.go` file.

- Basically the state request and response definition is defined at https://github.com/paritytech/substrate/blob/master/client/network/sync/src/schema/api.v1.proto

## Tests

N/A

## Issues

- Closes https://github.com/ChainSafe/gossamer/issues/4116

### CODE REVIEW

File: scripts/retrieve_state/retrieve_state.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 ChainSafe Systems (ON)
```
Comment (timwu20) [inline scripts/retrieve_state/retrieve_state.go]: unneeded file?

**Review by jimjbrettj — APPROVED** [review]
lgtm

**Review by dimartiro — APPROVED** [review]
LGTM and good refactor too!
